R/format_data.R

```diff
@@ -1602,21 +1619,44 @@ fmt_time <- function(data,
     data = data,
     columns = {{ columns }},
     rows = {{ rows }},
-    fns = list(
-      default = function(x) {
+    fns = num_fmt_factory_multi(
+      pattern = pattern,
+      use_latex_math_mode = FALSE,
+      format_fn = function(x, context) {
 
+        # If the incoming values are strings that adequately represent time
+        # values, then prepend with the `1970-01-01` dummy date to create an
+        # input that will works with `strftime()`
+        if (all(is_string_time(x))) {
+          x <- paste("1970-01-01", x)
+        }
+
+        # Convert incoming values to POSIXlt but provide a friendly error
+        # if the values cannot be parsed by `as.POSIXlt()`
         time <-
-          ifelse(grepl("^[0-9]*?\\:[0-9]*?", x), paste("1970-01-01", x), x) %>%
-          strftime(format = time_format_str)
+          tryCatch(
+            as.POSIXlt(x),
+            error = function(cond) {
+              stop(
+                "One or more of the provided date/time/date-time values are invalid.",
+                call. = FALSE
+              )
+            }
+          )
 
-        if (time_style %in% 3:5) {
-          time <- time %>% tidy_gsub(., "^0", "")
+        # Format the date string using `strftime()`
+        time <- strftime(time, format = time_format_str)
+
+        # Perform several cosmetic changes to the formatted time
+        if (grepl("%P$", time_format_str)) {
+
+          time <- gsub("^0", "", time)
+          time <- gsub(" 0([0-9])", " \\1", time)
+          time <- gsub("pm$", "PM", time)
+          time <- gsub("am$", "AM", time)
```

Comment:
Can we add word boundaries around "am" and "pm"?

Reply:
Now changed.